torch/fx/graph.py

```diff
@@ -533,17 +545,18 @@ def process_outputs(self, out: Any) -> Any:
         assert(self.pytree_info.out_spec is not None)
         return pytree.tree_unflatten(out, self.pytree_info.out_spec)
 
-    def gen_fn_def(self, free_vars, maybe_return_annotation):
+    def gen_fn_def(self, free_vars : List[_ParamDescr], maybe_return_annotation):
         if self.pytree_info is None:
             return super().gen_fn_def(free_vars, maybe_return_annotation)
         function_args = self.pytree_info.orig_args
-        has_orig_self = (function_args[0] == 'self')
+        has_orig_self = (function_args[0].name == 'self')
         if has_orig_self:
-            free_vars.insert(0, 'self')
+            free_vars.insert(0, _ParamDescr('self', '', ''))
         function_definition = super().gen_fn_def(function_args[:], maybe_return_annotation)
         if len(free_vars) > 0:  # pytree has placeholders in it
             function_definition += f"""
-    {', '.join(free_vars)}, = fx_pytree.tree_flatten_spec([{', '.join(function_args)}], self._in_spec)"""
+    {', '.join(var.name for var in free_vars)}, = fx_pytree.tree_flatten_spec\
```

Comment:
why trailing comma before `=` ?

Reply:
That forces it to always be a tuple even if there's 1 element